subsys/bluetooth/audio/tmap.c

```diff
@@ -155,10 +160,61 @@ static ssize_t read_role(struct bt_conn *conn,
 static struct bt_gatt_attr svc_attrs[] = { BT_TMAS_SERVICE_DEFINITION };
 static struct bt_gatt_service tmas;
 
+static bool valid_tmap_role(enum bt_tmap_role role)
+{
+	if (role == 0 || (role & TMAP_ALL_ROLES) != role) {
+		LOG_DBG("Invalid role %d", role);
+	}
+
+	if ((role & BT_TMAP_ROLE_CG) != 0 && !IS_ENABLED(CONFIG_BT_TMAP_CG_SUPPORTED)) {
+		LOG_DBG("Device does not support the CG role");
+
+		return false;
+	}
+
+	if ((role & BT_TMAP_ROLE_CT) != 0 && !IS_ENABLED(CONFIG_BT_TMAP_CT_SUPPORTED)) {
+		LOG_DBG("Device does not support the CY role");
+
+		return false;
+	}
+
+	if ((role & BT_TMAP_ROLE_UMS) != 0 && !IS_ENABLED(CONFIG_BT_TMAP_UMS_SUPPORTED)) {
+		LOG_DBG("Device does not support the UMS role");
+
+		return false;
+	}
+
+	if ((role & BT_TMAP_ROLE_UMR) != 0 && !IS_ENABLED(CONFIG_BT_TMAP_UMR_SUPPORTED)) {
+		LOG_DBG("Device does not support the UMR role");
+
+		return false;
+	}
+
+	if ((role & BT_TMAP_ROLE_BMS) != 0 && !IS_ENABLED(CONFIG_BT_TMAP_BMS_SUPPORTED)) {
+		LOG_DBG("Device does not support the BMS role");
+
+		return false;
+	}
+
+	if ((role & BT_TMAP_ROLE_BMR) != 0 && !IS_ENABLED(CONFIG_BT_TMAP_BMR_SUPPORTED)) {
+		LOG_DBG("Device does not support the BMR role");
+
+		return false;
+	}
+
+	return true;
+}
+
 int bt_tmap_register(enum bt_tmap_role role)
 {
 	int err;
 
+	CHECKIF(!valid_tmap_role(role)) {
```

Comment:
This is not specific for this PR, but also shows up other places where `CHECKIF` is used, and will be compiler dependent:
If the KCONFIG option `NO_RUNTIME_CHECKS` is set to 'y' a compiler may emit the following warning: `static bool valid_tmap_role defined but not used`

suggestions for fixing this?

Reply:
Looking at the macro it is defined as

```c
#elif defined(CONFIG_NO_RUNTIME_CHECKS)
#define CHECKIF(...) \
	if (0)
#else
```

With `if (0)` the code is still compiled, but not linked, and should thus not have any issues. Will leave as is

Reply:
Yes, the code compiles and works fine as is, the only issue is that we get a compiler warning, that shouldn't be there.